core/src/test/java/org/frankframework/pipes/EscapePipeTest.java

```diff
@@ -51,6 +51,39 @@ public void testPartialEscapedMultiLineXmlInput() throws Exception { //double es
 		TestAssertions.assertEqualsIgnoreCRLF(expected, result.getResult().asString());
 	}
 
+	@Test
+	public void testPartialEscapedMultiLineXmlInputMultiOccurences() throws Exception {
+		pipe.setSubstringStart("<message>");
+		pipe.setSubstringEnd("</message>");
+		configureAndStartPipe();
+
+		PipeRunResult result = doPipe(getResource("multi-line-partial-escaped-multi-occurences.xml"));
+		String expected = getResource("multi-line-partial-escaped-multi-occurences.result").asString();
+		TestAssertions.assertEqualsIgnoreCRLF(expected, result.getResult().asString());
+	}
+
+	@Test
+	public void nothingGetsEscapedWhenNoSubstringEndConfigured() throws Exception {
+		pipe.setSubstringStart("<message>");
+		pipe.setSubstringEnd("");
+		configureAndStartPipe();
+
+		PipeRunResult result = doPipe(getResource("no-substring-configured.xml"));
+		String expected = getResource("no-substring-configured.result").asString();
```

Comment:
Input and output seem identical?

In general in such a situation I think you can just reuse the other input file and assert that input and output are identical, but however I think this test is actually invalid -- see other comment above.

Reply:
I thought this is exactly what we want.
When no substringEnd is configured we don't know between what we would like to escape so we escape nothing